AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginAuthCognitoTests/AWSDataStoreAuthBaseTest.swift

```diff
@@ -203,144 +197,75 @@ extension AWSDataStoreAuthBaseTest {
         }
-
-        return result
     }
 
-    func getUserSub() -> String {
-        let retrieveUserSubCompleted = expectation(description: "retrieve userSub completed")
-        var resultOptional: String?
-        _ = Amplify.Auth.fetchAuthSession(listener: { event in
-            switch event {
-            case .success(let authSession):
-                guard let cognitoAuthSession = authSession as? AuthCognitoIdentityProvider else {
-                    XCTFail("Could not get auth session as AuthCognitoIdentityProvider")
-                    return
-                }
-                switch cognitoAuthSession.getUserSub() {
-                case .success(let userSub):
-                    resultOptional = userSub
-                    retrieveUserSubCompleted.fulfill()
-                case .failure(let error):
-                    XCTFail("Failed to get auth session \(error)")
-                }
-            case .failure(let error):
-                XCTFail("Failed to get auth session \(error)")
-            }
-        })
-        wait(for: [retrieveUserSubCompleted], timeout: TestCommonConstants.networkTimeout)
-        guard let result = resultOptional else {
-            XCTFail("Could not get userSub for user")
-            return ""
+    func isSignedIn() async throws -> Bool {
+        do {
+            let authSession = try await Amplify.Auth.fetchAuthSession()
+            return authSession.isSignedIn
+        } catch {
+            fatalError("Failed to get auth session of signed In user with error: \(error)")
         }
-
-        return result
     }
-
-    func getIdentityId() -> String {
-        let retrieveIdentityCompleted = expectation(description: "retrieve identity completed")
-        var resultOptional: String?
-        _ = Amplify.Auth.fetchAuthSession(listener: { event in
-            switch event {
-            case .success(let authSession):
-                guard let cognitoAuthSession = authSession as? AuthCognitoIdentityProvider else {
-                    XCTFail("Could not get auth session as AuthCognitoIdentityProvider")
-                    return
-                }
-                switch cognitoAuthSession.getIdentityId() {
-                case .success(let identityId):
-                    resultOptional = identityId
-                    retrieveIdentityCompleted.fulfill()
-                case .failure(let error):
-                    XCTFail("Failed to get auth session \(error)")
-                }
-            case .failure(let error):
-                XCTFail("Failed to get auth session \(error)")
+    
+   func getUserSub() async throws -> String {
+       do {
+           let authSession = try await Amplify.Auth.fetchAuthSession(options: nil)
+           let cognitoAuthSession = authSession as? AuthCognitoIdentityProvider
+           guard let userSub = try cognitoAuthSession?.getUserSub().get() else {
+               throw "Failed to get userSub"
```

Comment:
let's fail the test using XCTFail here, i think from our debugging session, it seems a lot easier to pinpoint the problem when it hits XCTFail
```suggestion
               XCTFail("Missing userSub() from session")
               throw "Failed to get userSub"
```